demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/demos/getbalance/GetBalanceFragment.kt

```diff
@@ -60,13 +61,20 @@ class GetBalanceFragment : BaseDemoFragment<FragmentGetBalanceBinding>() {
         binding.shield.apply {
             setOnClickListener {
                 lifecycleScope.launch {
-                    sharedViewModel.synchronizerFlow.value?.shieldFunds(
+                    val usk =
                         DerivationTool.getInstance().deriveUnifiedSpendingKey(
                             seed,
                             network,
                             Account.DEFAULT
                         )
-                    )
+                    sharedViewModel.synchronizerFlow.value?.let { synchronizer ->
+                        synchronizer.proposeShielding(usk.account, Zatoshi(100000))?.let { it1 ->
```

Comment:
This is 0.001 ZEC, currently worth 3.4 US cents. Because the minimum ZIP 317 fee is 10000 zats or 0.0001 ZEC, up to 10% (maybe only 5%, not sure) of the value could be lost in fees if people are sending you micropayments. I suggest raising the threshold by a factor of 10 (until the price increases, at least). Also, this threshold should be defined as a constant somewhere rather than being a magic number.

Reply:
Oh this is just the demo app. Hmm, but real wallets may copy this code.